engine/src/main/java/com/agiletec/aps/system/services/group/cache/GroupManagerCacheWrapper.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2018-Present Entando Inc. (http://www.entando.com) All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package com.agiletec.aps.system.services.group.cache;
+
+import com.agiletec.aps.system.common.AbstractCacheWrapper;
+import com.agiletec.aps.system.exception.ApsSystemException;
+import com.agiletec.aps.system.services.group.Group;
+import com.agiletec.aps.system.services.group.IGroupDAO;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.Cache;
+
+/**
+ * @author E.Santoboni
+ */
+public class GroupManagerCacheWrapper extends AbstractCacheWrapper implements IGroupManagerCacheWrapper {
+
+	private static final Logger _logger = LoggerFactory.getLogger(GroupManagerCacheWrapper.class);
+
+	private static enum Action {
+		ADD, UPDATE, DELETE
+	}
+
+	@Override
+	public void initCache(IGroupDAO groupDAO) throws ApsSystemException {

```

Comment:
Probably too late now but it seems like a lot of this could be in the Abstract implementation with a parameterized type. The init, release, insert and get calls are the same across all of the implementations.  Could be something like `public void initCache(ICacheableDAO<T> cacheDAO) ...` where T is an implementation of something that caches. Or there could be an ICacheable interface that marks assets as cacheable and expose some kind of update/process methods for custom logic. Then all of this boilerplate lives in the Abstract wrapper and the custom calls like `manageGroups` below could be a managed call to the implementation.

Reply:
It's exactly the same thing I thought a couple of days ago. I was thinking of building the next wrappers in the same way and standardizing those already created, so as to modify them (all together) from the perspective you have described.